design/baremetal-operator/configmap_for_host_inspection_data.md

```diff
@@ -0,0 +1,193 @@
+and BareMetalHost must be in the same namespace.
+
+- BareMetalHost is created
+- BareMetalHost is in inspecting state
+- Operator creates a HostData with the same name and namespace as BareMetalHost, fills out
+    the HostData with the host inspection data and sets ownerReferences
+- Provisioning is requested
+- Pivoted to a target cluster, operator restores host inspection data from HostData
+- Deprovisioning is requested
+- BareMetalHost deletion requested, operator deletes the HostData and BareMetalHost
+
+The spec of HostData:
+
+```go
+// Adding HardwareDetails struct here for visibility. During implementation,
+// we don't have to duplicate it, but use from baremetalhost_types.go.
+type HardwareDetails struct {
+
+    SystemVendor HardwareSystemVendor `json:"systemVendor,omitempty"`
+    Firmware     Firmware             `json:"firmware,omitempty"`
+    RAMMebibytes int                  `json:"ramMebibytes,omitempty"`
+    NIC          []NIC                `json:"nics,omitempty"`
+    Storage      []Storage            `json:"storage,omitempty"`
+    CPU          CPU                  `json:"cpu,omitempty"`
+    Hostname     string               `json:"hostname,omitempty"`
+
+}
+
+type HostDataSpec struct {
+    // The hardware discovered on the host.
+    HardwareDetails *HardwareDetails `json:"hardware,omitempty"`
+}
+```
+
+### Risks and Mitigations
+
+Existing deployments will require installation of the HostData CRD and necessary
+RBAC permissions when bumping the operator version. This needs to be clearly
+documented.
+
+### Work Items
+
+- Add HostData API
+- Modify the operator code
+- Clean up statusAnnotation related code from BMO and CAPM3
+- Add unit tests
+- Update documentation
+
+### Dependencies
+
+None
+
+### Test Plan
+
+Unit tests should be added. Integration test doesn't require any change.
+
+### Upgrade / Downgrade Strategy
+
+From a user perspective, this feature doesn't introduce any breaking changes
+to existing deployments, because `status.hardware` field will stay until is
+decided otherwise and this feature is only an addition to the current APIs.
+For now, there will be two copies of inspection data available, in the
+`spec.hardware` of HostData and `status.hardware` of BareMetalHost.
+
+### Version Skew Strategy
+
+To mitigate version skew issues, we plan to keep BaremetalHost and HostData
+synchronized by bi-directional copying of inspection data.
+
+## Drawbacks
+
+None
+
+## Alternatives
+
+- Use k8s built in objects like ConfigMap/Secret. From security point of view, although,
+    it is possible to restrict permissions to edit ConfigMaps (only to those
+    storing inspection data) based on the name (i.e. `resourceNames`), but it will add
+    complexity to the operator.
+- Increase metadata global limit. However, this does not always work as it is hard
```

Comment:
Adding filters to filter-out unneeded inspection data?

Reply:
Thanks for suggestion, but I'm afraid that even if we add a filter to ignore unnecessary fields we still will hit the issue, because the size of the unnecessary fields is relatively small and instead it is `status.hardware` that we care about and which might grow a lot depending on the hardware.

Reply:
valid point